server/src/services/media.service.ts

```diff
@@ -245,29 +247,31 @@ export class MediaService extends BaseService {
       } else {
         // did not extract or extracted preview is smaller than target size,
         // convert a full-sized thumbnail from original instead
-        extractedPath = StorageCore.getImagePath(asset, AssetPathType.EXTRACTED, image.preview.format);
-        // const orientation = asset.exifInfo?.orientation ? Number(asset.exifInfo.orientation) : undefined;
-        await this.mediaRepository.generateThumbnail(
+        extractedPath = StorageCore.getImagePath(asset, AssetPathType.CONVERTED, image.preview.format);
+        // unset size to get full size
+        decodeOptions.size = undefined;
```

Comment:
Better to handle this as a ternary when `decodeOptions` is instantiated.

Reply:
This `size` cannot be handled on `decodeOptions` init, as `useExtracted` was not determined yet.
the `size` should be set when we extract from a non-RAW asset,
i.e. the `imageIsRaw || useExtracted` is false, 
or the opposite - unset when `imageIsRaw && useExtracted` is true.